src/utils/JiraTicketStatusFetcher.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace staabm\PHPStanTodoBy\utils;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\ClientException;
+use Psr\Http\Message\StreamInterface;
+
+final class JiraTicketStatusFetcher implements TicketStatusFetcher
+{
+    private const API_VERSION = 2;
+
+    private Client $client;
+    private string $credentials;
+
+    public function __construct(string $host, string $credentialsFilePath)
+    {
+        $credentials = self::getCredentials($credentialsFilePath);
+
+        $this->credentials = base64_encode($credentials);
+        $this->client = new Client([
```

Comment:
can we utulize some kind of caching and/or cache revalidation?

I remember a [dedicated jira api client](https://github.com/lesstif/php-jira-rest-client) which supports it, so we could do similar stuff I think. I don't say we neeed the jira api client but maybe we can take inspiration from it. I am also fine with adopting the jira api client.. wdyt?

Reply:
I wonder that maybe phpstan's cache is enough? I don't know exactly how phpstan's cache works but I assume that if it's not invalidated altogether for some reason (e.g. configuration change), then the same comment won't be analyzed twice if it doesn't change. This means that the API call is done only once.

If we wanted to use cache for Jira requests in our extension how do you think it should work? Should we introduce some parameter that developers could adjust to their needs (e.g. keep the status in cache for 5 minutes)? I tried to quickly analyze the client you mentioned but I don't see how it caches API calls. Can you please give me a hint where to look?

I like my dev tools to use as few dependencies as possible to minimize risks of conflicts, etc. Maybe it would be a good idea to not depend on jira client nor on guzzle and use low-level `ext-curl` and `ext-json` functions to make these calls? This should be quite simple to do and there would be less things to worry during the maintenance of the extension. If we add support in the future for more issue trackers, then using some client dependency for each of them could lead to quite complex dependency graph which isn't necessarily required for such a simple task as fetching a status from APIs.  What do you think?

Reply:
Agree. As simple as possible would be great. 

It seems I misremembered the cache thing with a client to a different api. I can't find it too :-)